cmd/gardener-extension-shoot-rsyslog-relp-admission/app/app.go

```diff
@@ -104,13 +102,6 @@ func NewAdmissionCommand(ctx context.Context) *cobra.Command {
 					return err
 				}
 				managerOptions.LeaderElectionConfig = sourceClusterConfig
-			} else {
-				// Restrict the cache for secrets to the configured namespace to avoid the need for cluster-wide list/watch permissions.
-				managerOptions.Cache = cache.Options{
-					ByObject: map[client.Object]cache.ByObject{
-						&corev1.Secret{}: {Namespaces: map[string]cache.Config{webhookOptions.Server.Completed().Namespace: {}}},
-					},
-				}
```

Comment:
@plkokanov we should consider using APIReader to read the Project Secret in the admission component: https://github.com/gardener/gardener-extension-registry-cache/blob/3b520080f3e608720ab1c3bf1aeaa24ac31886ff/pkg/admission/validator/cache/shoot.go#L124-L128

Anyways, this is out of scope of @timuthy 's PR.